Xamarin.PropertyEditing/ViewModels/PanelViewModel.cs

```diff
@@ -115,10 +115,18 @@ internal class PanelViewModel
 		public PanelViewModel (TargetPlatform targetPlatform)
 			: base (targetPlatform)
 		{
-			ArrangeModes = new List<ArrangeModeViewModel> {
-				new ArrangeModeViewModel (PropertyArrangeMode.Name, this),
-				new ArrangeModeViewModel (PropertyArrangeMode.Category, this)
-			};
+			if (targetPlatform == null)
+				throw new ArgumentNullException (nameof(targetPlatform));
+
+			var modes = new List<ArrangeModeViewModel> ();
+			if (targetPlatform.ArrangeModes == null || targetPlatform.ArrangeModes.Count == 0)
+				modes.Add (new ArrangeModeViewModel (PropertyArrangeMode.Name, this));
+			else {
+				for (int i = 0; i < targetPlatform.ArrangeModes.Count; i++)
```

Comment:
Not sure why this isn't a `foreach`. Or `modes.AddRange (targetPlatform.ArrangeModes.Select (p => new ArrangeViewModel (p, this)))`, for that matter.

Reply:
Well, in fact, this could be further simplified:

```
ArrangeModes = (targetPlatform.ArrangeModes == null || targetPlatform.ArrangeModes.Count == 0) ?
    new List<ArrangeModeViewModel> (new ArrangeModeViewModel (PropertyArrangeMode.Name, this)) :
    targetPlatform.ArrangeModes.Select (p => new ArrangeViewModel (p, this)).ToList();
```

Reply:
Force of habit. I don't really see LINQ as an improvement here, no need to invoke the cost.